### PR TITLE
Improve the build.sh script

### PR DESCRIPTION
**Pull request by NicolasCARPi**
* User /usr/bin/env instead of hardcoded path for bash
* Check if command is in $PATH instead of hardcoding it in the script
* Exit early if the command is not found
* Use modern $() syntax instead of backticks
* Group the cleanup commands together
* Merge the mkdir commands in one with -p flag
* Add -f to rm composer.lock command so it doesn't complain if file is
  not here
* Script is now passing shellcheck with no errors
* Add verbose flag to final move
* Add phar builds to .gitignore

### CODE REVIEW

**Comment by NicolasCARPi** [comment]
Here is how I have it on my computer:

* `~/.bin` and `./opt` are in `$PATH`
* `~/.bin` is for little script/programs I write
* `~/.opt` is for userspace programs I download (like `psalm`, `composer`, `hugo`, `terraform`, ...)

I'd recommend doing the same for `phar-composer.phar`, adding it in `.opt` ;)

Also, I don't know if you're french, but in english, there is no space before `:`, `!` or `?`!

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 articles/
+builds/*.phar
```
Comment (eric-therond) [inline .gitignore]: The goal was to push dev versions to users who don't want to run the build script as suggested in the [README.md](https://github.com/designsecurity/progpilot/blob/master/README.md).

Two possibilities:
- remove `builds/*.phar` from `.gitignore` as before
- change the [README.md](https://github.com/designsecurity/progpilot/blob/master/README.md) to remove the link to [builds](./builds/) folder and suggest instead the execution of ./build.sh, something like that:
> Download the latest phar archive in [releases](https://github.com/designsecurity/progpilot/releases) folder (or [build](./build.sh) your own phar archive).

**Comment by NicolasCARPi** [comment]
I went with the change in the README because phar archives are (binary) artifacts that can be built from the source code and as such must not be commited to the repository.

See the commit message for the changes, and the [generated README here](https://github.com/deltablot/progpilot/tree/build#progpilot).

I believe my changes make the installation/configuration/usage processes more clear to the average user.

Side note: before the Installation, a Description section would be welcome IMHO.

**Comment by eric-therond** [comment]
LGTM
(I will take care of additional changes in the new README like formatting stuff like that).

Thank you a lot for this PR 

PS: yes I am french ;)

Eric

**Pull request merged**